src/Microsoft.DocAsCode.Build.TableOfContents/TocResolver.cs

```diff
@@ -48,6 +49,53 @@ private TocItemInfo ResolveItemCore(TocItemInfo wrapper, Stack<FileAndType> stac
             }
 
             var item = wrapper.Content;
+
+            // HomepageUid and Uid is deprecated, unified to TopicUid
+            if (string.IsNullOrEmpty(item.TopicUid))
+            {
+                if (!string.IsNullOrEmpty(item.Uid))
+                {
+                    item.TopicUid = item.Uid;
+                    item.Uid = null;
+                }
+                else
+                {
+                    if (!string.IsNullOrEmpty(item.HomepageUid))
+                    {
+                        item.TopicUid = item.HomepageUid;
+                        item.HomepageUid = null;
+                    }
+                }
+            }
+            // Homepage is deprecated, unified to TopicHref
+            if (!string.IsNullOrEmpty(item.Homepage))
+            {
+                if (string.IsNullOrEmpty(item.TopicHref))
+                {
+                    item.TopicHref = item.Homepage;
+                }
+                else
+                {
+                    Logger.LogWarning($"Homepage is deprecated in TOC. Homepage {item.Homepage} is overwritten with topicHref {item.TopicHref}");
+                }
+            }
+
+            // TocHref supports 2 forms: absolute path and local toc file.
```

Comment:
Absolute path is only used in breadcrumb toc for DOCS, which is not a common case. So only mention the local toc file usage in documentation.

---
In reply to: [73344695](https://github.com/dotnet/docfx/pull/537#discussion_r73344695) [](ancestors = 73344695)